ang/civicase/MasonryGrid.js

```diff
@@ -0,0 +1,118 @@
+(function (_, $, angular) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseMasonryGrid', function ($timeout) {
+    return {
+      controller: 'civicaseMasonryGridController',
+      link: civicaseMasonryGridLink
+    };
+
+    /**
+     * Masonry Grid link function.
+     *
+     * @param {Object} $scope the directive's scope.
+     * @param {Object} $element a reference to the directive's element.
+     * @param {Object} attrs a map of attributes associated to the element.
+     * @param {Object} masonryGrid a reference to the directive's controller.
+     */
+    function civicaseMasonryGridLink ($scope, $element, attrs, masonryGrid) {
+      (function init () {
+        $timeout(function () {
+          appendMasonryContainers();
+          arrangeGridItems();
+          $scope.$on('civicaseMasonryGrid::updated', arrangeGridItems);
+        });
+      })();
+
+      /**
+       * Appends the masonry containers used for splitting the grid items.
+       */
+      function appendMasonryContainers () {
+        _.times(2).forEach(function () {
+          $('<div></div>')
```

Comment:
honestly i have the feeling that instead of this DOM manipulation, you should have simply used a template for this directive
```html
<!-- directive.html -->
<div class="civicase__summary-tab-tile civicase__masonry-grid__container">
</div>
<div class="civicase__summary-tab-tile civicase__masonry-grid__container">
</div>
```
Then you would add `transclude: true` to the directive
```js
module.directive('civicaseMasonryGrid', function ($timeout) {
  return {
    transclude: true,
    controller: 'civicaseMasonryGridController',
    link: civicaseMasonryGridLink
  };
```
and use [`$transcludeFn`](https://triangular.io/blog/transclusion-transcludefn/) to do all the dom manipulation you need

right now it seems that you're not using the directive fully

Reply:
will investigate more about this!

Reply:
@AkA84 I tried to work a bit with `$transcludeFn` but I don't think it's useful at all. It gives you a reference to the whole transcluded element instead of individual grid items and they are not even ready by the time I want to manipulate them into different columns.

Reply:
@reneolivo i really think that you're manually doing something that the directive already does out of the box, but the time constraint is what it is, so let's keep things as they are